crates/api_common/src/utils.rs

```diff
@@ -835,13 +840,101 @@ fn limit_expire_time(expires: DateTime<Utc>) -> LemmyResult<Option<DateTime<Utc>
   }
 }
 
+pub async fn process_markdown(
+  text: &str,
+  slur_regex: &Option<Regex>,
+  context: &LemmyContext,
+) -> LemmyResult<String> {
+  let text = remove_slurs(text, slur_regex);
+  let (text, links) = markdown_rewrite_image_links(text);
+  RemoteImage::create(&mut context.pool(), links).await?;
+  Ok(text)
+}
+
+pub async fn process_markdown_opt(
+  text: &Option<String>,
+  slur_regex: &Option<Regex>,
+  context: &LemmyContext,
+) -> LemmyResult<Option<String>> {
+  match text {
+    Some(t) => process_markdown(t, slur_regex, context).await.map(Some),
+    None => Ok(None),
+  }
+}
+
+pub(crate) async fn proxy_image_link(
+  link: Url,
+  image_proxy: bool,
```

Comment:
Its necessary for tests. I could read the value from settings but there is no way to overwrite it from tests. Name is identical to the setting so that seems the most clear.

Reply:
How about creating a wrapper version, something like:

```rust
/// Should only be used for tests
fn proxy_image_link_wrapper(context, force_use_image_proxy: bool) {
  ...current code
}

fn proxy_image_link(context) {
  fn proxy_image_link_wrapper(context, context.pictrs....image_proxy.unwrap_or_default() || force_use_image_proxy)
}
```

Reply:
nm, I'll get it in a PR